setup.py

```diff
@@ -24,7 +22,10 @@
         'console_scripts': [('productivity = productivity:command_line')]
     },
     install_requires=[
-        'pymodbus>=2.4.0,<3',
+        'pymodbus>=2.4.0,<3; python_version == "3.7"',
+        'pymodbus[serial]>=2.4.0; python_version == "3.8"',
```

Comment:
This will not be needed in `3.0.1`.  See https://github.com/riptideio/pymodbus/issues/1128.  Perhaps I should just wait until then (Python 3.11 should be released, also)